torch/lib/TensorLib/CUDAGenerator.cpp

```diff
@@ -0,0 +1,44 @@
+#ifdef TENSORLIB_CUDA_ENABLED
```

Comment:
Untested. THC differs because there is a single generator associated with THCState, as opposed to generators being passed to functions. This code is sourced from THPP.